### ISSUE TITLE
react/no-typos emit false error.

### ISSUE THREAD

**Issue by Aliaksei-Martsinkevich** [yannickcr/eslint-plugin-react#1738]
This rule fails only on external proptypes declaration. Using static class field propTypes works fine.
```JavaScript
// validators.js
import PropTypes from 'prop-types';

export const validator = PropTypes.shape({ ... });

// Component.jsx
import { validator } from 'validators';
function Component({ value }) { ... }
Component.propTypes = {
  value: validator.isRequired,
//                ^-----------^   - Typo in declared prop type: isRequired (react/no-typos)
};
```

**Comment by ljharb** [comment]
hmm, that should definitely work.

**Comment by JustFly1984** [comment]
I get errors on ImmutablePropTypes.map and ImmutablePropTypes.list

```
Typo in declared prop type: map  react/no-typos
Typo in declared prop type: list  react/no-typos
```

Please help. Currently it breaks my build, so I will mute this rule until you fix it

**Comment by ljharb** [comment]
@JustFly1984 please file a separate issue for that.

**Comment by Cordazar** [comment]
This is really bothersome when you have lots and lots of stateless components.

**Comment by alexzherdev** [comment]
I've looked into this, but I'm unable to reproduce with the following test case (which produces no errors):
```jsx
import { validator } from 'validators';
function MyComponent(props) { return (<div>{props.myProp}</div>) }
MyComponent.propTypes = {
  myProp: validator.isRequired
};
```
